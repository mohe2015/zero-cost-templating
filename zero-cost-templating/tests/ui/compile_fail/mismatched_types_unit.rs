#![deny(warnings)]
#![feature(async_closure, async_iterator, coroutines, gen_blocks, noop_waker)]

extern crate alloc;

use alloc::borrow::Cow;

use zero_cost_templating::yields;
use zero_cost_templating_macros::template_stream;

#[template_stream("test.html.hbs")]
pub gen fn test() -> Cow<'static, str> {
    let template = yields!(test0());
    let page_title = Cow::from("thetitle");
    let template = yields!(template.page_title0(page_title));
    let csrf_token = Cow::from("thetoken");
    yields!(template.csrf_token1(csrf_token));
}

pub fn main() {}
