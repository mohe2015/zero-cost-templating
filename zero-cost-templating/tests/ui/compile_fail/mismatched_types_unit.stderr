error[E0308]: mismatched types
  --> tests/ui/compile_fail/mismatched_types_unit.rs:20:5
   |
12 | pub gen fn test() -> Cow<'static, str> {
   |                            ----------------- expected `()` because of return type
...
20 |     yields!(template.test_template4())
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `()`, found `Template<TestTemplate5, (), ()>`
   |
   = note: expected unit type `()`
                 found struct `Template<TestTemplate5, (), ()>`
   = note: this error originates in the macro `yields` (in Nightly builds, run with -Z macro-backtrace for more info)
